### PR TITLE
[UWP] Inline Text Runs 

### PR DESCRIPTION
**Pull request by RebeccaAnne**
#2377

### CODE REVIEW

File: source/uwp/Renderer/idl/AdaptiveCards.Rendering.Uwp.idl [hunk 1]
```diff
@@ -702,12 +721,120 @@ AdaptiveNamespaceStart
...
+#ifdef ADAPTIVE_CARDS_WINDOWS
+        uuid(D81768ED-D724-4494-B141-A9B3F2613EFD),
+        contract(InternalContract, 1),
+        internal
+#else
+        uuid(09BBE759-0754-441D-B808-3CE2C188D521)
+#endif
+    ] interface IAdaptiveTextRun : IInspectable requires IAdaptiveTextElement
+    {
+        [propget] HRESULT SelectAction([out, retval] IAdaptiveActionElement * *value);
```
Comment (paulcam206) [inline source/uwp/Renderer/idl/AdaptiveCards.Rendering.Uwp.idl]: >* * [](start = 76, length = 3)

nitpick: spacing here and below

File: source/uwp/Renderer/lib/AdaptiveParagraph.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            return E_INVALIDARG;
+        }
+
+        GenerateInlinesProjection(sharedParagraph->GetInlines(), m_inlines.Get());
+
+        return S_OK;
+    }
+    CATCH_RETURN;
+
+    HRESULT AdaptiveParagraph::get_Inlines(ABI::Windows::Foundation::Collections::IVector<ABI::AdaptiveNamespace::IAdaptiveInline*>** inlines)
```
Comment (paulcam206) [inline source/uwp/Renderer/lib/AdaptiveParagraph.cpp]: >ABI [](start = 43, length = 3)

missing SAL here

File: source/uwp/Renderer/lib/AdaptiveRichTextBlock.cpp [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    HRESULT AdaptiveRichTextBlock::put_HorizontalAlignment(ABI::AdaptiveNamespace::HAlignment alignment)
+    {
+        m_horizontalAlignment = alignment;
+        return S_OK;
+    }
+
+    HRESULT AdaptiveRichTextBlock::get_Paragraphs(
+        ABI::Windows::Foundation::Collections::IVector<ABI::AdaptiveNamespace::AdaptiveParagraph*>** paragraphs)
```
Comment (paulcam206) [inline source/uwp/Renderer/lib/AdaptiveRichTextBlock.cpp]: >ABI [](start = 8, length = 3)

missing SAL

File: source/uwp/Renderer/lib/AdaptiveRichTextBlockRenderer.cpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    HRESULT AdaptiveRichTextBlockRenderer::RuntimeClassInitialize() noexcept try
+    {
+        return S_OK;
+    }
+    CATCH_RETURN;
+
+    HRESULT AdaptiveRichTextBlockRenderer::Render(_In_ IAdaptiveCardElement* cardElement,
+                                              _In_ IAdaptiveRenderContext* renderContext,
+                                              _In_ IAdaptiveRenderArgs* renderArgs,
+                                              _COM_Outptr_ ABI::Windows::UI::Xaml::IUIElement** result) noexcept try
```
Comment (paulcam206) [inline source/uwp/Renderer/lib/AdaptiveRichTextBlockRenderer.cpp]: is this the spacing from clang?

Comment (RebeccaAnne) [reply source/uwp/Renderer/lib/AdaptiveRichTextBlockRenderer.cpp]: nope, fixed

---
In reply to: [266631607](https://github.com/Microsoft/AdaptiveCards/pull/2552#discussion_r266631607) [](ancestors = 266631607)

File: source/uwp/Renderer/lib/AdaptiveTextElement.h [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+            sharedCardElement->SetFontStyle(static_cast<AdaptiveSharedNamespace::FontStyle>(m_fontStyle));
+            sharedCardElement->SetTextSize(static_cast<AdaptiveSharedNamespace::TextSize>(m_textSize));
+            sharedCardElement->SetTextWeight(static_cast<AdaptiveSharedNamespace::TextWeight>(m_textWeight));
+            sharedCardElement->SetTextColor(static_cast<AdaptiveSharedNamespace::ForegroundColor>(m_foregroundColor));
+
+            std::string text;
+            RETURN_IF_FAILED(HStringToUTF8(m_text.Get(), text));
+            sharedCardElement->SetText(text);
+
+            std::string language;
```
Comment (paulcam206) [inline source/uwp/Renderer/lib/AdaptiveTextElement.h]: >td::string language; [](start = 13, length = 20)

nitpick: this should probably be declared inside the following scope

**Pull request merged**